### DISCUSSION TITLE
Using KSCAN touch controller with LVGL

### DISCUSSION THREAD

**Discussion by nategreco** [zephyrproject-rtos/zephyr#53971] [Q&A]
I've got an application that uses LVGL for GUI development and wish to connect my touch input to the LVGL objects (buttons, sliders, etc), but am having a bit of trouble.

First - I'm struggling to find much documentation on the topic at all.  Searching every which way on docs.zephyrproject.org only turns up the basic kscan.h API documentation, unrelated to LVGL.  I have implemented this and it works very well and is responsive, however it cannot be used to manipulate the LVGL objects.

The only info I've found on using KScan with LVGL is in the LVGL sample.  It seems the following configuration lines do the magic I need:
CONFIG_LV_Z_POINTER_KSCAN=y
CONFIG_LV_Z_POINTER_KSCAN_DEV_NAME="FT5336"

But the performance is appalling! I see several seconds of delay and inconsistent screen response.  Also, my GUI has a (unwanted) scrollbar at the bottom and the screen seems to jump around when I touch any of the text labels.

Given the above, any of the following would be helpful:
- Any documentation at all regarding how to connect KScan touch_controller devices to LVGL objects
- Any suggestions on how to improve LVGL performance (is there somewhere I should change a thread priority?  More frequent calls to lv_task_handler()?)
- How to remove the scrollbar and disable screen sliding panning

Other background info:
- Nordic nRF52840
- Zephyr 3.1.99 (nRF Connectect v2.1.0 SDK)
- 3.5" 320x480 TFT capacitive touch screen (ILI9488 + FT5446)

**Comment by pdgendt** [comment]
Hi @nategreco, it might require a manifest update, see #49577 that fixes this

**Reply by nategreco** [reply]
Good find @pdgendt!  It's just that one line change to zephyr/lvgl.c, right?

I'm actually using the Nordic nRF Connect SDK so I can't just pull down a newer version of Zephyr, but did make that change local but it seems it didn't have a noticeable impact.  I'll look into it a bit more.
